components/common/src/listen_ctl_addr.rs

```diff
@@ -53,20 +51,24 @@ impl FromStr for ListenCtlAddr {
     type Err = Error;
 
     fn from_str(val: &str) -> Result<Self> {
-        Ok(ListenCtlAddr(SocketAddr::from_str(val)?))
+        Ok(val.parse::<SocketAddr>()?.into())
     }
 }
 
-impl ToSocketAddrs for ListenCtlAddr {
-    type Iter = option::IntoIter<SocketAddr>;
+impl fmt::Display for ListenCtlAddr {
+    fn fmt(&self, f: &mut fmt::Formatter) -> result::Result<(), fmt::Error> {
+        write!(f, "{}", self.0)
+    }
+}
 
-    fn to_socket_addrs(&self) -> io::Result<Self::Iter> {
-        self.0.to_socket_addrs()
+impl From<SocketAddr> for ListenCtlAddr {
+    fn from(socket_addr: SocketAddr) -> Self {
+        ListenCtlAddr(socket_addr)
     }
 }
 
-impl fmt::Display for ListenCtlAddr {
-    fn fmt(&self, f: &mut fmt::Formatter) -> result::Result<(), fmt::Error> {
-        write!(f, "{}", self.0)
+impl Into<SocketAddr> for ListenCtlAddr {
+    fn into(self) -> SocketAddr {
+        self.0
```

Comment:
Based on [this API guideline](https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#c-conv-traits) we probably shouldn't implement this. 

(sorry, I know that's the opposite of what I suggested yesterday... I am full of fail :frowning:)

Reading [further](https://rust-lang-nursery.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv), if we really need this functionality, we should probably have an `into_inner()` method that returns `self.0` instead.

Reply:
I think `AsRef<SockAddr>` is what we want. See https://github.com/habitat-sh/habitat/pull/5950/files#r240687807.